### ISSUE TITLE
OSError at master startup

### ISSUE THREAD

**Issue by matdrapeau** [saltstack/salt#20872]
```
[ERROR   ] Exception [Errno 1] Operation not permitted: '/var/cache/salt/master/roots/mtime_map' occurred in file server update
Traceback (most recent call last):
  File "/Users/mdrapeau/projects/utils/salt/salt/daemons/masterapi.py", line 239, in fileserver_update
    fileserver.update()
  File "/Users/mdrapeau/projects/utils/salt/salt/fileserver/__init__.py", line 311, in update
    self.servers[fstr]()
  File "/Users/mdrapeau/projects/utils/salt/salt/fileserver/roots.py", line 124, in update
    with salt.utils.fopen(mtime_map_path, 'rb') as fp_:
  File "/Users/mdrapeau/projects/utils/salt/salt/utils/__init__.py", line 1057, in fopen
    os.chown(path, uid, gid)
OSError: [Errno 1] Operation not permitted: '/var/cache/salt/master/roots/mtime_map'
```

**Comment by rallytime** [comment]
@matdrapeau Can you tell us what version of salt your running (post the output from `salt --versions-report`) as well as what OS you are running your master on?

**Comment by matdrapeau** [comment]
@rallytime the reworked commit is done on 2015.2 and the system where the manager is running is MacOSx

**Comment by jfindlay** [comment]
@cachedout, another unintended consequence of globally chowning all files touched by salt.utils.fopen.

**Comment by jfindlay** [comment]
Except that I'm quite wrong and @matdrapeau is right.  I had incorrectly assumed that salt.utils.fopen chowned all files.  It only chowns files if you explicitly give it uid, gid, according to #20715.

I'm going to revert my erroneous fix in favor of #20878.

**Comment by cachedout** [comment]
Looks like we already have a fix in for this. Closing.
